python/hail/methods/statgen.py

```diff
@@ -204,6 +204,729 @@ def linreg(dataset, ys, x, covariates=[], root='linreg', block_size=16):
+
+    return cleanup(MatrixTable(jds))
+
+@handle_py4j
+@typecheck(ds=MatrixTable,
+           kinshipMatrix=KinshipMatrix,
+           y=Expression,
+           x=Expression,
+           covariates=listof(Expression),
+           global_root=strlike,
+           va_root=strlike,
+           run_assoc=bool,
+           use_ml=bool,
+           delta=nullable(numeric),
+           sparsity_threshold=numeric,
+           n_eigs=nullable(integral),
+           dropped_variance_fraction=(nullable(float)))
+def lmmreg(ds, kinshipMatrix, y, x, covariates=[], global_root="lmmreg_global", va_root="lmmreg",
+           run_assoc=True, use_ml=False, delta=None, sparsity_threshold=1.0,
+           n_eigs=None, dropped_variance_fraction=None):
+    r"""Use a kinship-based linear mixed model to estimate the genetic component
+    of phenotypic variance (narrow-sense heritability) and optionally test each
+    variant for association.
+
+    Examples
+    --------
+    Compute a :class:`.KinshipMatrix`, and use it to test variants for
+    association using a linear mixed model:
+
+    >>> ds = hc.read("data/example_lmmreg.vds")
+    >>> kinship_matrix = methods.rrm(ds.filter_rows(ds.useInKinship)['GT'])
+    >>> lmm_ds = methods.lmmreg(ds, kinship_matrix, ds.pheno, ds.GT.num_alt_alleles(), [ds.cov1, ds.cov2])
+
+    Notes
+    -----
+    Suppose the variant dataset saved at :file:`data/example_lmmreg.vds` has a
+    Boolean variant-indexed field `useInKinship` and numeric or Boolean
+    sample-indexed fields `pheno`, `cov1`, and `cov2`. Then the
+    :func:`.lmmreg` function in the above example will execute the following
+    four steps in order:
+
+    1) filter to samples in given kinship matrix to those for which
+       `ds.pheno`, `ds.cov`, and `ds.cov2` are all defined
+    2) compute the eigendecomposition :math:`K = USU^T` of the kinship matrix
+    3) fit covariate coefficients and variance parameters in the
+       sample-covariates-only (global) model using restricted maximum
+       likelihood (`REML`_), storing results in a global field under
+       `lmmreg_global`
+    4) test each variant for association, storing results in a row-indexed
+       field under `lmmreg`
+
+    .. _REML: https://en.wikipedia.org/wiki/Restricted_maximum_likelihood
+
+    This plan can be modified as follows:
+
+    - Set `run_assoc` to :obj:`False` to not test any variants for association,
+      i.e. skip Step 5.
+    - Set `use_ml` to :obj:`True` to use maximum likelihood instead of REML in
+      Steps 4 and 5.
+    - Set the `delta` argument to manually set the value of :math:`\delta`
+      rather that fitting :math:`\delta` in Step 4.
+    - Set the `global_root` argument to change the global annotation root in
+      Step 4.
+    - Set the `va_root` argument to change the variant annotation root in
+      Step 5.
+
+    :func:`.lmmreg` adds 9 or 13 global annotations in Step 4, depending on
+    whether :math:`\delta` is set or fit.
+
+    .. list-table::
```

Comment:
This table also wraps kind of badly, but I think that's just a problem with big tables in general. You might be able to see a little more of the `Value` field by moving the `lmmreg_global` root out of the table, but that might be more confusing?